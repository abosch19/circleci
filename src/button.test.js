import React from 'react'
import { shallow, mount } from 'enzyme'
import Button from './button'

describe('Button test', () => {
    it('render', () => {
        const wrapper = shallow(<Button/>);

        expect(wrapper).toMatchSnapshot()
    })

    it('text is modified', () => {
        const wrapper = mount(<Button/>);

        const button = wrapper.find('button')

        button.simulate('click')
        button.simulate('click')

        expect(wrapper.state().timesClicked).toEqual(2)
        expect(button.text()).toEqual('2 times clicked!')
    })

    it('props onClick works', () => {
        const onClick = jest.fn()
        const wrapper = mount(<Button onClick={onClick} />);

        const button = wrapper.find('button')

        button.simulate('click')
        button.simulate('click')

        expect(onClick).toHaveBeenCalledTimes(2)

    })
})