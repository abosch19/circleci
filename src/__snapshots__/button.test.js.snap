// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test render 1`] = `
<button
  onClick={[Function]}
>
  0
   times clicked!
</button>
`;
